File: scaffolder/templates/new-product/skeleton-java/src/test/java/com/example/app/HealthControllerTest.java
package com.example.app;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.test.web.servlet.MockMvc;

/** Asserts the liveness/readiness endpoint the platform probes returns 200 {"status":"ok"}. */
@WebMvcTest(HealthController.class)
class HealthControllerTest {

    @Autowired
    private MockMvc mvc;

    @Test
    void healthzReturnsOk() throws Exception {
        mvc.perform(get("/healthz"))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$.status").value("ok"));
    }
}

File: scaffolder/templates/new-product/skeleton-ruby/test/controllers/health_controller_test.rb
require "test_helper"

# Asserts the liveness/readiness endpoint the platform probes returns 200 {"status":"ok"}.
class HealthControllerTest < ActionDispatch::IntegrationTest
  test "GET /healthz returns 200 ok" do
    get "/healthz"
    assert_response :success
    assert_equal "ok", JSON.parse(response.body)["status"]
  end
end
